soh/soh/Enhancements/enemyrandomizer.cpp

```diff
@@ -0,0 +1,174 @@
+    ACTOR_EN_RD,        // Redead, Gibdo
+    ACTOR_EN_SW,        // Skullwalltula
+    ACTOR_EN_FD,        // Flare Dancer
+    ACTOR_EN_KAREBABA,  // Withered Deku Baba
+    ACTOR_EN_RR,        // Like-Like
+    ACTOR_EN_NY,        // Spike (rolling enemy)
+    ACTOR_EN_IK,        // Iron Knuckle
+    ACTOR_EN_TUBO_TRAP, // Flying pot
+    ACTOR_EN_FZ,        // Freezard
+    ACTOR_EN_WEIYER,    // Stinger (Water)
+    ACTOR_EN_HINTNUTS,  // Hint deku scrubs
+    ACTOR_EN_WF,        // Wolfos
+    ACTOR_EN_SKB,       // Stalchild
+    ACTOR_EN_CROW       // Guay
+};
+
+extern "C" enemyEntry GetRandomizedEnemy(void) {
+    return randomizedEnemySpawnTable[rand() % RANDOMIZED_ENEMY_SPAWN_TABLE_SIZE];
+}
+
+extern "C" uint8_t IsEnemyFoundToRandomize(int actorId = 0, int param = 0) {
+
+    uint8_t isEnemyFound = 0;
+
+    for (int i = 0; i < sizeof(enemiesToRandomize) / sizeof(enemiesToRandomize[0]); i++) {
+        if (actorId == enemiesToRandomize[i]) {
+            isEnemyFound = 1;
+        }
+    }
+
+    if (isEnemyFound) {
+        switch (actorId) {
+            // Only randomize the main component of Electric Tailparasans, not the tail segments they spawn
+            case ACTOR_EN_TP:
+                if (param != -1) {
+                    return 0;
+                }
+                break;
+            // Only randomize the initial deku scrub actor (single and triple attack), not the flower they spawn.
+            case ACTOR_EN_DEKUNUTS:
+                if (param != -256 && param != 768) {
+                    return 0;
+                }
+                break;
+            // Only randomize initial floormaster actor (it can split and does some spawning on init)
+            case ACTOR_EN_FLOORMAS:
+                if (param != 0) {
+                    return 0;
+                }
+                break;
+            // Only randomize initial egg spawn, not the enemy that comes out of the egg
+            case ACTOR_EN_GOMA:
+                if (param != 0 && param != 6 && param != 8) {
+                    return 0;
+                }
+                break;
+            // Only randomize Skullwalltulas, not Golden Skulltulas
+            case ACTOR_EN_SW:
+                if (param != 0) {
+                    return 0;
+                }
+                break;
+            // Don't randomize Nabooru because it'll break cutscenes and progression
+            case ACTOR_EN_IK:
+                if (param == 1280) {
+                    return 0;
+                }
+                break;
+            // Only randomize the intitial spawn of the huge jellyfish. It spawns another copy when hit with a sword.
+            case ACTOR_EN_VALI:
+                if (param != -1) {
+                    return 0;
+                }
+                break;
+            default:
+                break;
+        }
+
+        // Enemy found.
+        return 1;
+    }
+
+    // If no enemy is found, don't randomize the actor.
+    return 0;
+}
```

Comment:
minor optimization, this way the array isn't iterated through to the end after finding a match

also changed the `for` to use the `ARRAY_COUNT` macro

```suggestion
extern "C" uint8_t IsEnemyFoundToRandomize(int actorId = 0, int param = 0) {
    for (int i = 0; i < ARRAY_COUNT(enemiesToRandomize); i++) {
        if (actorId == enemiesToRandomize[i]) {
            switch (actorId) {
                // Only randomize the main component of Electric Tailparasans, not the tail segments they spawn
                case ACTOR_EN_TP:
                    if (param != -1) {
                        return 0;
                    }
                    break;
                // Only randomize the initial deku scrub actor (single and triple attack), not the flower they spawn.
                case ACTOR_EN_DEKUNUTS:
                    if (param != -256 && param != 768) {
                        return 0;
                    }
                    break;
                // Only randomize initial floormaster actor (it can split and does some spawning on init)
                case ACTOR_EN_FLOORMAS:
                    if (param != 0) {
                        return 0;
                    }
                    break;
                // Only randomize initial egg spawn, not the enemy that comes out of the egg
                case ACTOR_EN_GOMA:
                    if (param != 0 && param != 6 && param != 8) {
                        return 0;
                    }
                    break;
                // Only randomize Skullwalltulas, not Golden Skulltulas
                case ACTOR_EN_SW:
                    if (param != 0) {
                        return 0;
                    }
                    break;
                // Don't randomize Nabooru because it'll break cutscenes and progression
                case ACTOR_EN_IK:
                    if (param == 1280) {
                        return 0;
                    }
                    break;
                // Only randomize the intitial spawn of the huge jellyfish. It spawns another copy when hit with a sword.
                case ACTOR_EN_VALI:
                    if (param != -1) {
                        return 0;
                    }
                    break;
                default:
                    return 1;
            }
        }
    }

    // If no enemy is found, don't randomize the actor.
    return 0;
}
```

Reply:
Good catch, that'll work. I still need a return 1 after the switch statement though.

Reply:
> I still need a return 1 after the switch statement though.

that's what the `default:` is doing

Reply:
oh wait, because the breaks won't get to the default, nevermind

Reply:
actually, let me take another stab at this, just sec

Reply:
alright @aMannus i updated the suggested change comment

Reply:
That's definitely a lot cleaner, thank you! Seems to work fine too :)